store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedRemoveGraphHandler.java

```diff
@@ -35,10 +40,34 @@
 public class FederatedRemoveGraphHandler implements OutputOperationHandler<RemoveGraph, Boolean> {
     @Override
     public Boolean doOperation(final RemoveGraph operation, final Context context, final Store store) throws OperationException {
+        boolean removed;
+        Set<Class<? extends Operation>> supportedOperations = getSupportedOperations(context, (FederatedStore) store);
         try {
-            return ((FederatedStore) store).remove(operation.getGraphId(), context.getUser(), operation.isUserRequestingAdminUsage());
+            removed = ((FederatedStore) store).remove(operation.getGraphId(), context.getUser(), operation.isUserRequestingAdminUsage());
         } catch (final Exception e) {
             throw new OperationException("Error removing graph: " + operation.getGraphId(), e);
         }
+        if (removed) {
+            Set<Class<? extends Operation>> updatedSupportedOperations = getSupportedOperations(context, (FederatedStore) store);
+            if (!supportedOperations.equals(updatedSupportedOperations)) {
+                supportedOperations.removeAll(updatedSupportedOperations);
+                for (final Class<? extends Operation> removedOperation : supportedOperations) {
+                    if (((FederatedStore) store).getExternallySupportedOperations().contains(removedOperation)) {
```

Comment:
else? what happens?
is that illegal state? do we care?

Reply:
The if statement is needed because it checks that the operations that are no longer supported by subgraphs are not operations that should still be supported by the federated store itself.
Otherwise if you have 2 subgraphs, an Accumulo store and a federated store, and you removed the sub federated store, all of the federated operation handlers would be deleted. This is why the `getExternallySupportedOperations` flag is required.

Reply:
So to answer "else what happens?": If one store is added and then removed, the `GetElementsHandler` would be detected as removed from the sub graphs, but we see that this is not in the `externallySupportedOperations` as the federated store can support this itself, so we do not delete the handler.

Reply:
Can you review this logic.

1) AddGraph(FedStore) // has AddGraph
2) FederatedAddGraphParentHandler -> addExternallySupportedOperations() // AddGraph is added.
3) RemoveGraphHandler line44 -> getSupportedOperations // includes AddGraph from (1)
4) RemoveGraphHandler line51 -> getSupportedOperations // does NOT include AddGraph
5) line 52 not equal
6) line 53 supported list has AddGraph (it wasn't removed)
7) line 55 externallySupportedOperations contains AddGraph 
8) line 56 remove AddGraph from current FederatedStore.

FederatedStore no supports AddGraphs.

Reply:
See new test `shouldKeepFederatedStoreSupportedOperations`

Reply:
`AddGraph` is never added to `externallySupportedOperations` because in the [`FederatedAddGraphHandlerParent::addGenericHandler`](https://github.com/gchq/Gaffer/blob/dce22c074b9fcc89cb6f4b2c9cc8e193217e8066/store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java#L94) it only gets added when `!store.isSupported(supportedOperation)`

Reply:
thanks. I think I'm satisfied by this.